### DISCUSSION TITLE
Uvicorn overrides existing signal handlers.

### DISCUSSION THREAD

**Discussion by tchaton** [Kludex/uvicorn#1708] [Potential Issue]
From my experimentation,

I have found uvicorn overrides existing handlers.

In PyTorch Lightning, we resolve this using HandlerCompose: [connectors](https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/trainer/connectors/signal_connector.py#L22).

It would be great if uvicorn could resolve this bug.

**Comment by haf** [comment]
ping: this is an issue

If you're running a main computation, and you want SIGINT to work on that main asyncio loop:

<img width="682" alt="image" src="https://user-images.githubusercontent.com/193115/219065248-e725571d-7e6c-4c22-aba9-8054daf058f5.png">

it doesn't anymore:

<img width="581" alt="image" src="https://user-images.githubusercontent.com/193115/219065379-aefb6bc2-2a83-4095-bdca-139283294568.png">

**Comment by haf** [comment]
Solution with aiohttp:

<img width="694" alt="image" src="https://user-images.githubusercontent.com/193115/219070681-f207985d-5d20-4b8c-b117-d1ee6dd98211.png">

then later:

```python
await runner.shutdown()
```

**Comment by haf** [comment]
Ping
